lib/ssl/src/ssl_handshake.erl

```diff
@@ -2209,24 +2203,24 @@ bad_key(#{algorithm := ecdsa}) ->
     unacceptable_ecdsa_key.
 
 cert_status_check(_,
-                  #{ocsp_state := #{ocsp_stapling := true,
-                                    ocsp_expect := stapled}},
+                  #{stapling_state := #{configured := true,
+                                        status := stapled}},
                   _VerifyResult, _, _) ->
     %% OCSP staple will now be checked by
     %% ssl_certificate:verify_cert_extensions/2 in ssl_certificate:validate
     valid;
 cert_status_check(OtpCert,
-                  #{ocsp_state := #{ocsp_stapling := false}} = SslState,
+                  #{stapling_state := #{configured := false}} = SslState,
                   VerifyResult, CertPath, LogLevel) ->
     maybe_check_crl(OtpCert, SslState, VerifyResult, CertPath, LogLevel);
 cert_status_check(_OtpCert,
-                  #{ocsp_state := #{ocsp_stapling := true,
-                                    ocsp_expect := undetermined}},
+                  #{stapling_state := #{configured := true,
+                                        status := undetermined}},
                   _VerifyResult, _CertPath, _LogLevel) ->
     {bad_cert, {revocation_status_undetermined, not_stapled}};
```

Comment:
check if this clause is dead code. openssl_stapling_SUITE works fine without it.

Reply:
probably need, but current tests don't verify it.
- not_negotiated(aka no staple) - is tested in `staple_missing` testcase
- not_received(aka undetermined) - can't be tested easily and currently it is not covered
I'm leaving that clause

Reply:
Or maybe there is a missing test case!  I think this looks like we want to fail if no staple was received .

Reply:
This might be tricky to test. You would need TLS server negotiated status_request but not sending the response ...